### PR TITLE
Binary decimal conversion

### PR DESCRIPTION
**Pull request by Muna-S**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by TagwaHashim — COMMENTED** [review]
very good job muna.
just check the py formatting and py test.

File: solutions/binary_decimal_conversion.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+"""
+A module for converting between binary and decimal numbers. 
+Created by Muna Sattouf on 29/12/2024
+Completed on 1/1/2025
+"""
+def DecimalToBinary(decimal: int)-> str:
```
Comment (FalaqMajeed) [inline solutions/binary_decimal_conversion.py:6]: Great work on the function! I noticed that the function name isn't in snake case, which is the standard convention in Python. Could you please update it to snake case? For example, you could rename it to something like decimal_to_binary for better readability and consistency.

File: solutions/binary_decimal_conversion.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    This function takes a decimal number from command line as paramter,
+    and returns its binary representation.
+    
+    Example: DecimalToBinary('binary', 392)
```
Comment (FalaqMajeed) [inline solutions/binary_decimal_conversion.py:11]: Having more than three examples is a great way to help others understand the functionality of your function. It makes the code much clearer and easier to follow

File: solutions/binary_decimal_conversion.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    >>> 110001000
+    
+    """
+    if not isinstance(decimal, int):
```
Comment (FalaqMajeed) [inline solutions/binary_decimal_conversion.py:15]: One small suggestion: adding a space between the function and the assert statements could improve readability and make the logic clearer.

File: solutions/tests/test_binary_decimal_conversion.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+contains test cases for the solution code.
+
+Created by Muna Sattouf on January 2, 2024
+Completed on January 2, 2024
+"""
+import unittest
+from binary_decimal_conversion import DecimalToBinary, BinaryToDecimal, binarydecimalconversion
+
+class TestBinaryDecimalConversion(unittest.TestCase):
+
```
Comment (FalaqMajeed) [inline solutions/tests/test_binary_decimal_conversion.py:12]: Your implementation is looking great! Could you please add a docstring to the class? It will help others (and future you) understand the purpose and functionality of the class at a glance.

**Review by FalaqMajeed — COMMENTED** [review]
Would it be possible for you to go through the changes as soon as you can? We're running a bit short on time

File: solutions/tests/test_binary_decimal_conversion.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+"""
+Module: test_binary_decimal_conversion
+contains test cases for the solution code.
+
+Created by Muna Sattouf on January 2, 2024
+Completed on January 2, 2024
+"""
+import unittest
+from binary_decimal_conversion import decimal_to_binary, binary_decimal_conversion, binary_to_decimal
```
Comment (sashour82) [inline solutions/tests/test_binary_decimal_conversion.py:9]: Why you are importing from multiple files?!!!

Comment (Muna-S) [reply solutions/tests/test_binary_decimal_conversion.py:9]: I am importing from only one file - however I am importing 3 function from that file. 
The file is binary_decimal_conversion.

File: solutions/binary_decimal_conversion.py (hunk 2 above)
Comment (Muna-S) [reply solutions/binary_decimal_conversion.py:15]: added an example to this function + there is an example to the other function

File: solutions/binary_decimal_conversion.py (hunk 1 above)
Comment (Muna-S) [reply solutions/binary_decimal_conversion.py]: Changed to snake case

File: solutions/tests/test_binary_decimal_conversion.py (hunk 4 above)
Comment (Muna-S) [reply solutions/tests/test_binary_decimal_conversion.py]: docstring added

**Review by TagwaHashim — APPROVED** [review]
Good job.I will merge now
